### PR TITLE
Embedded Auth for Android

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **What's changed?** Main setup, sample app setup, user sign in and user sign out and various edits
- **Is this PR related to a Monolith release?** Tuscany

### Resolves:

* [OKTA-413947](https://oktainc.atlassian.net/browse/OKTA-413947)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/index.md [hunk 1]
```diff
@@ -94,15 +78,11 @@ For the sample app, use the following scopes: `openid`, `profile`, `offline_acce
 
 After you complete the configurations, run the sample app:
 
-* [Run the SDK sample app](/docs/guides/oie-embedded-sdk-run-sample/aspnet/main/)
-* [Run the Widget sample app](/docs/guides/oie-embedded-widget-run-sample/aspnet/main/)
+* [Run the SDK sample app](/docs/guides/oie-embedded-sdk-run-sample/-/main/)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/index.md:81]:
```suggestion
* [Run the SDK sample app](/docs/guides/oie-embedded-sdk-run-sample/aspnet/main/)
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/index.md [hunk 2]
```diff
@@ -94,15 +78,11 @@ For the sample app, use the following scopes: `openid`, `profile`, `offline_acce
...
+* [Run the Widget sample app](/docs/guides/oie-embedded-widget-run-sample/-/main/)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/index.md:82]:
```suggestion
* [Run the Widget sample app](/docs/guides/oie-embedded-widget-run-sample/aspnet/main/)
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/clientsecret.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+### Client secret
+
+For the sample app, use the **Client secret** for the app integration that you created in [Create new application](/docs/guides/oie-embedded-common-org-setup/java/main/#create-a-new-application). To find this value:
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/clientsecret.md:3]: We shouldn't use client secrets for Android.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/sdkforyourapp.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+1. In your Okta Org, [create a new application integration](/docs/guides/oie-embedded-common-org-setup/java/main/#create-a-new-application) for your app.
+1. Ensure that you have all the [software requirements](#software-requirements).
+
+#### Step 2: Import packages and add framework
+
+The embedded authentication with SDK sample apps use the Android framework with the Okta Identity Engine Java SDK. Import the Okta API packages as well as any Android packages that you need.
+
+```java
+package com.okta.spring.example;
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/sdkforyourapp.md:13]:
```suggestion
package com.okta.android.example;
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/softwarerequirements.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+* [Android Studio](https://developer.android.com/studio) 4.2.x or later
+* [JDK 8](https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html) or later
+* [Apache Maven](https://maven.apache.org/download.cgi) 3.6.x or later
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/softwarerequirements.md:3]: We don't need this one.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/softwarerequirements.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+    Include the following Okta Identity Engine Java SDK dependency for Apache Maven:
+
+    ```xml
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/softwarerequirements.md:7]: We should replace this with a gradle dependency block.
`implementation 'com.okta.idx.sdk: okta-idx-java-api:$okta_sdk_version'`

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+ "Sign in screenshot")
+
+</div>
+
+### Step 2: Authenticate user credentials
+
+When the user initiates the sign-in process, your app needs to create an [AuthenticationOptions](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/model/AuthenticationOptions.java) object and set its `username` and `password` properties to the values entered by the user. Send this object to the
+[IDXAuthenticationWrapper](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java) `authenticate` method to authenticate the user.
+
+```java
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md:21]:
```suggestion
```kotlin
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+Depending on [AuthenticationResponse](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/response/AuthenticationResponse.java)'s [AuthenticationStatus](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/model/AuthenticationStatus.java) value, you need to handle the response accordingly:
+
+#### Success status
+
+For a successful sign-in response (`AuthenticationStatus.Success`), use the [AuthenticationResponse](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/response/AuthenticationResponse.java)'s `getTokenResponse` method to retrieve the token and then process the authenticated user in the app.
+
+For example:
+
+```java
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md:49]:
```suggestion
```kotlin
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    ...
+```
+
+#### Other status
+
+You need to handle other returned [AuthenticationStatus](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/model/AuthenticationStatus.java) cases if the user didn't sign in successfully.
+
+For example:
+
+```java
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md:73]:
```suggestion
```kotlin
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+When the sign-out request is initiated, create the following flow:
+
+1. Obtain the access token from the active session state.
+
+1. Call `revokeToken` on [IDXAuthenticationWrapper](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java), passing in the access token obtained from the previous step.
+
+1. Invalidate the current session.
+
+```java
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md:23]:
```suggestion
```kotlin
```

**Review by JayNewstrom — COMMENTED** [review]
It would be nice to have a preamble stating that the android sample uses the java sdk, which works with android and backend java or something along those lines.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/softwarerequirements.md [hunk 11]
```diff
@@ -1,15 +1,10 @@
...
 
     Include the following Okta Identity Engine Java SDK dependency for Apache Maven:
 
-    ```xml
-    <dependency>
-        <groupId>com.okta.idx.sdk</groupId>
-        <artifactId>okta-idx-java-api</artifactId>
-        <version>${okta.sdk.version}</version>
-    </dependency>
+    ```java
```
Comment (JayNewstrom) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/android/softwarerequirements.md:6]:
```suggestion
    ```groovy
```

**Review by JayNewstrom — APPROVED** [review]

**Review by jaynewstrom-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/ios/sdkforyourapp.md [hunk 12]
```diff
@@ -1,3 +1,5 @@
+After you run the sample app and explore its available use cases, you can begin to integrate the SDK and/or the widget into your own app. To get started follow these steps:
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/ios/sdkforyourapp.md:1]:
```suggestion
After you run the sample app and explore its available use cases, you can begin to integrate the SDK and/or the widget into your own app. To get started, follow these steps:
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+                return ProceedTransition.ErrorTransition(response.errors)
+            }
+            return null
+    }
+    ...
+```
+
+#### Other status
+
+You need to handle other returned [AuthenticationStatus](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/model/AuthenticationStatus.java) cases if the user didn't sign in successfully.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md:69]:
```suggestion
You need to handle the other returned [AuthenticationStatus](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/model/AuthenticationStatus.java) cases if the user didn't sign in successfully.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+                        unsupportedPolicy()
+                    }
+                }
+                else -> unsupportedPolicy()
+            }
+        }
+    ...
+```
+
+### Step 4: Get user profile information-optional
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md:108]: hyphen before optional or optional in parentheses (optional)?

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/summaryofsteps.md [hunk 15]
```diff
@@ -0,0 +1,9 @@
+## Summary of steps
+
+This use case entails building out a simple UI to capture the username, password, and authenticating the credentials with Okta using the SDK.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/summaryofsteps.md:3]:
```suggestion
This use case explains how to build a simple UI to capture the username and password, and authenticate the credentials with Okta using the SDK.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/index.md [hunk 16]
```diff
@@ -29,23 +27,19 @@ You need to complete the following configurations before you continue with this
 
 ### Step 1: Complete steps in Set up your Okta org for password factor only use cases
 
-If not already done, complete the steps described in the
-[Set up your Okta org (for password factor only use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-password-factor-only-use-cases).
+If not already done, complete the steps described in the [Set up your Okta org (for password factor only use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-password-factor-only-use-cases).
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/index.md:30]:
```suggestion
If not already done, complete the steps described in [Set up your Okta org (for password factor only use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-password-factor-only-use-cases).
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md [hunk 17]
```diff
@@ -0,0 +1,33 @@
+## Integration steps
+
+### Step 1: Create a sign out UI element
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md:3]:
```suggestion
### Step 1: Create a sign-out UI element
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+
+The first step is to create a link, button, or other similar UI element that allows the user to sign out of the app.
+
+<div class="common-image-format">
+
+![Sign out link](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-out-link.png "Sign out link")
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md:9]:
```suggestion
![Sign-out link](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-out-link.png "Sign-out link")
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+
+```kotlin
+   ...
+    fun signOut() {
+        transitionToForm(form = initialForm())
+    }
+    ...
+```
+
+### Step 3: Send user to the signed out page
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/integrationsteps.md:31]:
```suggestion
### Step 3: Send user to the signed-out page
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/summaryofsteps.md [hunk 20]
```diff
@@ -0,0 +1,9 @@
+## Summary of steps
+
+This use case signs a user out of the local app by revoking their session tokens.
+
+<div class="common-image-format">
+
+![Basic sign out sequence diagram for Java](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-out-seq-java.png "Basic sign out sequence diagram for Java]")
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-out/android/summaryofsteps.md:7]:
```suggestion
![Basic sign-out sequence diagram for Java](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-sign-out-seq-java.png "Basic sign-out sequence diagram for Java]")
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-overview/index.md [hunk 21]
```diff
@@ -7,9 +7,7 @@ title: Overview
 
 ## Start with a use case
 
-Using the sample app as your guide, walk through each of the use cases in
-this section to build your own app with the SDK. Each use case describes
-the steps required to integrate your app with the SDK.
+Using the sample app as your guide, walk through each of the use cases in this section to build your own app with the SDK. Each use case describes the steps required to integrate your app with the SDK.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-overview/index.md:10]:
```suggestion
With the sample app as your guide, walk through each of the use cases in this section to build your own app with the SDK. Each use case describes the steps that are required to integrate your app with the SDK.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md [hunk 22]
```diff
@@ -33,7 +33,7 @@ After your Okta org receives a successful Facebook sign-in request, your org red
 
 ### Step 5: Handle the callback from Okta
 
-Okta returns the Interaction code to the **Sign-in redirect URI** specified in the [create new application](/docs/guides/oie-embedded-common-org-setup/java/main/#step-4-create-new-application) step.
+Okta returns the Interaction code to the **Sign-in redirect URI** specified in the [create new application](/docs/guides/oie-embedded-common-org-setup/java/main/#create-a-new-application) step.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/java/integrationsteps.md:36]:
```suggestion
Okta returns the Interaction code to the **Sign-in redirect URI** that is specified in the [create new application](/docs/guides/oie-embedded-common-org-setup/java/main/#create-a-new-application) step.
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/nodejs/integrationsteps.md [hunk 23]
```diff
@@ -49,7 +49,7 @@ and must be identical. These locations are:
 
 For the sample application, the **RedirectURI** should be set to `http://localhost:8080/login/callback`
 
-Okta returns the interaction code to the **Sign-in redirect URI** specified in the [create new application step](/docs/guides/oie-embedded-common-org-setup/nodejs/main/#step-4-create-new-application) and is accessed in the sample app from `login.js`.
+Okta returns the interaction code to the **Sign-in redirect URI** specified in the [create new application step](/docs/guides/oie-embedded-common-org-setup/nodejs/main/#create-a-new-application) and is accessed in the sample app from `login.js`.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/nodejs/integrationsteps.md:52]:
```suggestion
Okta returns the interaction code to the **Sign-in redirect URI** that is specified in the [create new application step](/docs/guides/oie-embedded-common-org-setup/nodejs/main/#create-a-new-application) and is accessed in the sample app from `login.js`.
```

**Review by christinechoy-okta — COMMENTED** [review]
A few nits

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md (hunk 14 above)
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-basic-sign-in/android/integrationsteps.md:108]:
```suggestion
### Step 4: Get user profile information (optional)
```

**Review by christinechoy-okta — APPROVED** [review]

**Pull request merged**